src/main/java/net/mcreator/ui/MCreatorApplication.java

```diff
@@ -269,8 +269,22 @@ public WorkspaceSelector getWorkspaceSelector() {
 	 */
 	public MCreator openWorkspaceInMCreator(File workspaceFile) {
 		this.workspaceSelector.setCursor(new Cursor(Cursor.WAIT_CURSOR));
+
 		Workspace workspace = null;
 		try {
+			if (!Workspace.isWorkspaceLockedToVersion(workspaceFile)) {
+				int retval = JOptionPane.YES_OPTION;
+				if (!!Workspace.workspaceVersionMatchesCurrentVersion(workspaceFile)) {
+					retval = JOptionPane.showConfirmDialog(this.workspaceSelector,

```

Comment:
openWorkspaceInMCreator may not be called from EDT, requiring scheduling on the UI thread. Please handle threading properly here or this will cause lockups on macOS and similar platforms.

Check how other dialogs are handled in this file.